src/Repository/ShippingExportRepository.php

```diff
@@ -16,18 +16,38 @@
 
 class ShippingExportRepository extends EntityRepository implements ShippingExportRepositoryInterface
 {
+    public const NEW_STATE_PARAMETER = 'newState';
+
+    public const PENDING_STATE_PARAMETER = 'pendingState';
+
     public function createListQueryBuilder(): QueryBuilder
     {
         return $this->createQueryBuilder('o')
             ->leftJoin('o.shipment', 'shipment')
         ;
     }
 
+    /**
+     * @inheritdoc
+     */
     public function findAllWithNewState(): array
     {
+        trigger_deprecation('bitbag/shipping-export-plugin', '1.6', 'The "%s()" method is deprecated, use "ShippingExportRepository::findAllWithNewOrPendingState" instead.', __METHOD__);
+
         return $this->createQueryBuilder('o')
             ->where('o.state = :newState')
-            ->setParameter('newState', ShippingExportInterface::STATE_NEW)
+            ->setParameter(self::NEW_STATE_PARAMETER, ShippingExportInterface::STATE_NEW)
+            ->getQuery()
+            ->getResult()
+        ;
+    }
+
+    public function findAllWithNewOrPendingState(): array
```

Comment:
Can you write any integration tests for the repository methods?

Reply:
I've used behat to tests if it works. If you think there's still a need for integration just let me know 😄 .